paddle/fluid/operators/math/math_function.cu

```diff
@@ -33,6 +34,7 @@ using complex64 = paddle::platform::complex64;
 using complex128 = paddle::platform::complex128;
 
 template struct SetConstant<platform::CUDADeviceContext, platform::float16>;
+template struct SetConstant<platform::CUDADeviceContext, platform::bfloat16>;
```

Comment:
Are that changes needed here?  BF16 is enabled only for CPU.

Reply:
Unfortunately, because of fill_constant_op.h implementation it is needed to add this change.
But thanks to https://github.com/PaddlePaddle/Paddle/blob/e2d4ef9d4d8cfb278da3e3860fd503ed639a540c/paddle/fluid/operators/fill_constant_op.h#L109  only the CPU version will be enabled.

Reply:
To clarify, the point is that in `fill constant op` we can choose the place of calculations (DeviceContext) also using the "place_type" attribute, then in `fill_constant_op.h` we call `math::SetConstant` function where we set DeviceContext and date type. When I registered bfloat16 kernel, during compilation it is checked if math::SetConstant call is supported for all data types.  And I didn't find a better solution not to have to refactor the whole thing.

Reply:
LGTM